src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -284,18 +300,156 @@ ZeroInferRequest::ZeroInferRequest(const std::shared_ptr<ZeroInitStructsHolder>&
+        _tensorsData[stateOutputBufferName] = TensorData{_copyAllTensors.at(stateOutputBufferName)->data(),
+                                                         _copyAllTensors.at(stateOutputBufferName)->get_byte_size()};
+    }
+}
+
+void ZeroInferRequest::check_and_get_mem_data(std::string name, bool isInput, bool isState) {
+    std::string stateInputName;
+    bool setTensorData = false;
+
+    if (isState) {
+        stateInputName = READVALUE_PREFIX + name;
+        name = ASSIGN_PREFIX + name;
+    }
+
+    auto& tensor = _allTensors.at(name);
+    auto remoteTensor = std::dynamic_pointer_cast<ZeroRemoteTensor>(tensor);
+
+    if (remoteTensor == nullptr) {
+        auto& copyTensor = _copyAllTensors.at(name);
+        auto remoteCopyTensor = std::dynamic_pointer_cast<ZeroRemoteTensor>(copyTensor);
+
+        auto checkDataType = [&]() {
+            ze_memory_allocation_properties_t desc = {};
+            desc.stype = ZE_STRUCTURE_TYPE_MEMORY_ALLOCATION_PROPERTIES;
+            auto res = zeMemGetAllocProperties(_initStructs->getContext(), tensor->data(), &desc, nullptr);
+            if (res == ZE_RESULT_SUCCESS) {
+                switch (desc.type) {
+                case ZE_MEMORY_TYPE_HOST:
+                case ZE_MEMORY_TYPE_DEVICE:
+                case ZE_MEMORY_TYPE_SHARED:
+                    _copyAllTensors.at(name) = tensor;
+                    setTensorData = true;
+                    break;
+                case ZE_MEMORY_TYPE_UNKNOWN:
+                case ZE_MEMORY_TYPE_FORCE_UINT32:
+                    break;
+                }
+            }
+        };
+
+        if (remoteCopyTensor == nullptr) {
+            if (tensor->data() != copyTensor->data()) {
+                checkDataType();
+            }
+        } else {
+            checkDataType();
+
+            ov::Allocator allocator;
+            if (isInput && (_properties.flags & ZE_DEVICE_PROPERTY_FLAG_INTEGRATED)) {
+                allocator = zeroMemory::HostMemAllocator(_initStructs, ZE_HOST_MEM_ALLOC_FLAG_BIAS_WRITE_COMBINED);
+            } else {
+                allocator = zeroMemory::HostMemAllocator(_initStructs);
+            };
+
+            _copyAllTensors.at(name) = ov::make_tensor(tensor->get_element_type(), tensor->get_shape(), allocator);
+
+            setTensorData = true;
+        }
+
+        if (setTensorData == true) {
+            _tensorsData.at(name) =
+                TensorData{_copyAllTensors.at(name)->data(), _copyAllTensors.at(name)->get_byte_size()};
+            if (isState) {
+                _copyAllTensors.at(stateInputName) = _copyAllTensors.at(name);
+                _tensorsData.at(stateInputName) = TensorData{_copyAllTensors.at(stateInputName)->data(),
+                                                             _copyAllTensors.at(stateInputName)->get_byte_size()};
+            }
+            _createPipeline = true;
```

Comment:
Why does this function change state of `createPipeline` variable?

Reply:
In this way, we understand whether we must recreate the pipeline.